qutebrowser/mainwindow/prompt.py

```diff
@@ -617,6 +622,8 @@ def _set_fileview_root(self, path, *, tabbed=False):
             elif os.path.isdir(dirname) and not tabbed:
                 # Input like /foo/ba -> show /foo contents
                 path = dirname
+                if self._has_completions(dirname, basename):
```

Comment:
I wonder whether this check is needed - I think if I enter a directory name and it doesn't exist, it's fine to not do anything at all rather than to start cycling through items from the top - I think as a user, I'd only intend to cycle through all elements if I have no basename entered at all. What do you think?

Reply:
Yeah maybe :)
I was not totally sure what would be appropriate. I just thought this would keep more of the current behavior xD